modules/http/src/test/java/org/mule/module/http/functional/listener/HttpListenerUrlEncodedTestCase.java

```diff
@@ -128,6 +133,27 @@ public void urlEncodedEmptyParamsUrlEncodedContentTypeGenerateANullPayload() thr
         assertNullPayloadAndEmptyResponse(response);
     }
 
+    @Test
+    public void serverClosesConnectionAfterSendingData() throws Exception
+    {
+        // Apache Fluent doesn't fail while other clients such as curl, postman and this one do
+        AsyncHttpClientConfig asyncHttpClientConfig = getAsyncHttpClientConfig();
+        AsyncHttpClient asyncHttpClient = new AsyncHttpClient(new GrizzlyAsyncHttpProvider(asyncHttpClientConfig), asyncHttpClientConfig);
+        ListenableFuture<com.ning.http.client.Response> responseFuture = asyncHttpClient.
+                preparePost(getListenerUrl()).setBody("a=1&b=2").
+                addHeader("Content-Type", "application/x-www-form-urlencoded").execute();
+        com.ning.http.client.Response response = responseFuture.get();
+
+        assertThat(response.getStatusCode(), is(200));
+    }
+
+    private AsyncHttpClientConfig getAsyncHttpClientConfig()
```

Comment:
I don't think this is necessary. You can just inline it and get a simple config.